query/src/main/java/com/connexta/replication/query/TaskInfoImpl.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.query;
+
+import com.connexta.replication.api.data.MetadataInfo;
+import com.connexta.replication.api.data.OperationType;
+import com.connexta.replication.api.data.ResourceInfo;
+import com.connexta.replication.api.data.TaskInfo;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+public class TaskInfoImpl implements TaskInfo {
+
+  private String id;
+
+  private byte priority;
+
+  private OperationType operation;
+
+  private Instant lastModified;
+
+  @Nullable private ResourceInfo resourceInfo;
+
+  private Stream<MetadataInfo> metadatas;
+
+  public TaskInfoImpl() {
+    this.id = UUID.randomUUID().toString();
+  }
+
+  @Override
+  public String getId() {
+    return id;
+  }
+
+  @Override
+  public byte getPriority() {
+    return priority;
+  }
+
+  @Override
+  public OperationType getOperation() {
+    return operation;
+  }
+
+  @Override
+  public Instant getLastModified() {
+    return lastModified;
+  }
+
+  @Override
+  public Optional<ResourceInfo> getResource() {
+    return Optional.of(resourceInfo);
+  }
+
+  @Override
+  public Stream<MetadataInfo> metadatas() {
+    return metadatas;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public void setPriority(byte priority) {
+    this.priority = priority;
+  }
+
+  public void setOperation(OperationType operation) {
+    this.operation = operation;
+  }
+
+  public void setLastModified(Instant lastModified) {
+    this.lastModified = lastModified;
+  }
+
+  public void setResourceInfo(@Nullable ResourceInfo resourceInfo) {
+    this.resourceInfo = resourceInfo;
+  }
+
+  public void setMetadatas(Stream<MetadataInfo> metadatas) {
```

Comment:
✏️ Instead of set, how about `add` methods where you can pass in a stream of metadata, `MetadataInfo...`. This would make it easier on your clients and allow you to collect them internally on a list.